### PR TITLE
Support `.storybook/babel.config.js`

### PR DESCRIPTION
**Pull request by Hypnosphi**
Issue: https://github.com/storybooks/storybook/issues/2758#issuecomment-416367489

### CODE REVIEW

File: lib/core/src/server/loadCustomBabelConfig.js [hunk 1]
```diff
@@ -25,6 +26,17 @@ function loadFromPath(babelConfigPath) {
       logger.error(`=> Error parsing .babelrc file: ${e.message}`);
       throw e;
     }
+  } else {
+    try {
+      // eslint-disable-next-line import/no-dynamic-require,global-require
+      const jsConfig = require(path.resolve(babelConfigDir, 'babel.config.js'));
```
Comment (igor-dv) [inline lib/core/src/server/loadCustomBabelConfig.js]: I think it's better to use `serverRequire` to support also `babel.config.ts`

**Review by igor-dv — COMMENTED** [review]
I've started to work on presets and created these initial PRs #4027 #4043 (and also [presets-merge-babel-configs](https://github.com/storybooks/storybook/tree/core/presets-merge-babel-configs) branch). Considering this (and if everybody agrees with the API), maybe we can wait with this PR? If it will be released in v4, we won't be able to deprecate it in the next release

File: lib/core/src/server/loadCustomBabelConfig.js (hunk 1 above)
Comment (Hypnosphi) [reply lib/core/src/server/loadCustomBabelConfig.js]: Sure, I think I'll just close this one in favor of yours

Comment (igor-dv) [reply lib/core/src/server/loadCustomBabelConfig.js]: Cool =) Would love to hear your thoughts there

**Pull request closed without merging**